cranelift-codegen/src/legalizer/mod.rs

```diff
@@ -149,29 +148,43 @@ fn expand_cond_trap(
     // Split the EBB after `inst`:
     //
     //     trapnz arg
+    //     ..
     //
     // Becomes:
     //
-    //     brz arg, new_ebb
+    //     brz arg, new_ebb_resume
+    //     jump new_ebb_trap
+    //
+    //   new_ebb_trap:
     //     trap
-    //   new_ebb:
     //
+    //   new_ebb_resume:
+    //     ..
     let old_ebb = func.layout.pp_ebb(inst);
-    let new_ebb = func.dfg.make_ebb();
+    let new_ebb_trap = func.dfg.make_ebb();
+    let new_ebb_resume = func.dfg.make_ebb();
+
+    // Replace trap instruction by the inverted condition.
     if trapz {
-        func.dfg.replace(inst).brnz(arg, new_ebb, &[]);
+        func.dfg.replace(inst).brnz(arg, new_ebb_resume, &[]);
     } else {
-        func.dfg.replace(inst).brz(arg, new_ebb, &[]);
+        func.dfg.replace(inst).brz(arg, new_ebb_resume, &[]);
     }
 
+    // Add jump instruction after the inverted branch
     let mut pos = FuncCursor::new(func).after_inst(inst);
     pos.use_srcloc(inst);
+    pos.ins().jump(new_ebb_trap, &[]);
+    // Insert the new label and the unconditional trap terminator.
+    pos.insert_ebb(new_ebb_trap);
```

Comment:
Pre-existing: not sure if we should prefer `split_ebb` here. It seems that when `split_ebb` is used, we can then use `recompute_split_ebb` instead of `recompute_ebb`, which might do even less work. No strong preference here, current code is fine by me.

Reply:
`recompute_split_ebb` seems to only exists as part of the dominator code computation. I will leave this code as-is as I do not yet fully understand how the flow graph is stored nor manipulated.